contrib/platform/src/com/sun/jna/platform/win32/IPHlpAPI.java

```diff
@@ -0,0 +1,280 @@
+            "OutDiscards", "OutErrors", "OutUcastOctets", "OutMulticastOctets", "OutBroadcastOctets", "OutQLen" })
+    class MIB_IF_ROW2 extends Structure {
+        public long InterfaceLuid; // 64-bit union
+        public ULONG InterfaceIndex;
+        public GUID InterfaceGuid;
+        public char[] Alias = new char[IF_MAX_STRING_SIZE + 1];
+        public char[] Description = new char[IF_MAX_STRING_SIZE + 1];
+        public ULONG PhysicalAddressLength;
+        public UCHAR[] PhysicalAddress = new UCHAR[IF_MAX_PHYS_ADDRESS_LENGTH];
+        public UCHAR[] PermanentPhysicalAddress = new UCHAR[IF_MAX_PHYS_ADDRESS_LENGTH];
+        public ULONG Mtu;
+        public ULONG Type;
+        // enums
+        public int TunnelType;
+        public int MediaType;
+        public int PhysicalMediumType;
+        public int AccessType;
+        public int DirectionType;
+        // 8-bit structure
+        public byte InterfaceAndOperStatusFlags;
+        // enums
+        public int OperStatus;
+        public int AdminStatus;
+        public int MediaConnectState;
+        public GUID NetworkGuid;
+        public int ConnectionType;
+        public long TransmitLinkSpeed;
+        public long ReceiveLinkSpeed;
+        public long InOctets;
+        public long InUcastPkts;
+        public long InNUcastPkts;
+        public long InDiscards;
+        public long InErrors;
+        public long InUnknownProtos;
+        public long InUcastOctets;
+        public long InMulticastOctets;
+        public long InBroadcastOctets;
+        public long OutOctets;
+        public long OutUcastPkts;
+        public long OutNUcastPkts;
+        public long OutDiscards;
+        public long OutErrors;
+        public long OutUcastOctets;
+        public long OutMulticastOctets;
+        public long OutBroadcastOctets;
+        public long OutQLen;
+    }
+
+    /**
+     * The IP_ADDRESS_STRING structure stores an IPv4 address in dotted decimal
+     * notation. The IP_ADDRESS_STRING structure definition is also the type
+     * definition for the IP_MASK_STRING structure.
+     * 
+     * @see <A HREF=
+     *      "https://docs.microsoft.com/en-us/windows/desktop/api/iptypes/ns-iptypes-ip_address_string">IP_ADDRESS_STRING</A>
+     */
+    @FieldOrder({ "String" })
+    class IP_ADDRESS_STRING extends Structure {
+        // Null terminated string
+        // up to 3 chars (decimal 0-255) and dot
+        // ending with null
+        public byte[] String = new byte[16];
+    }
+
+    /**
+     * The IP_ADDR_STRING structure represents a node in a linked-list of IPv4
+     * addresses.
+     * 
+     * @see <A HREF=
+     *      "https://docs.microsoft.com/en-us/windows/desktop/api/iptypes/ns-iptypes-_ip_addr_string">IP_ADDR_STRING</A>
+     */
+    @FieldOrder({ "Next", "IpAddress", "IpMask", "Context" })
+    class IP_ADDR_STRING extends Structure {
+        public ByReference Next;

```

Comment:
Please qualify this to `IP_ADDR_STRING.ByReference` I had to look multiple times.